pkg/client/cache/listers.go

```diff
@@ -95,6 +95,10 @@ func (s *StoreToPodLister) Exists(pod *api.Pod) (bool, error) {
 	return exists, nil
 }
 
+// NodeConditionPredicate is a function that indicates if a given list of node conditions match the
```

Comment:
This comment is confusing because the argument to the function is not a list of node conditions. I would slightly re-word to say "NodeConditionPredicate is a function that indicates whether the given node's conditions meet some set of criteria defined by the function."